_posts/2024-07-03-oidc-cient-request.adoc

```diff
@@ -0,0 +1,440 @@
+
+```
+service/keycloak created
+route.route.openshift.io/keycloak created
+Warning: apps.openshift.io/v1 DeploymentConfig is deprecated in v4.14+, unavailable in v4.10000+
+deploymentconfig.apps.openshift.io/keycloak created.
+```
+It will take a few minutes for OpenShift to provision the Keycloak pod and its related resources.
+
+You can use the OpenShift CLI or the OpenShift web console, depending on your preference, to check if your Keycloak server has been provisioned.
+
+=== OpenShift CLI
+To make sure your Keycloak server has been provisioned using the OpenShift CLI, run:
+```
+oc get pods
+```
+
+After a little while, check for a message similar to the following message that indicates the pod is ready:
+```
+NAME                READY     STATUS      RESTARTS   AGE
+keycloak-1-deploy   0/1       Completed   0          1h
+keycloak-1-l9kdx    1/1       Running     0          1h
+```
+
+Once the Keycloak server has been provisioned, use the following command to find the URL for your Keycloak instance’s Admin Console:
+```
+KEYCLOAK_URL=https://$(oc get route keycloak --template='{{ .spec.host }}') &&
+echo "" &&
+echo "Keycloak Admin Console:   $KEYCLOAK_URL/admin" &&
+echo ""
+```
+
+=== OpenShift Web Console
+To make sure your Keycloak server has been provisioned using the OpenShift web console, navigate to the *Topology* view in the *Developer* perspective. You can click on your *keycloak* app to check its status. Once it is running, you can click on *Open URL* and then access Keycloak’s *Administration Console*.
+
+== Setting up Keycloak OpenID Provider
+
+. Log into the Keycloak Admin Console using the admin username and password you specified earlier. 
+. Next, create a realm called *myrealm* and register a client called *myclient* as follows: 
+*General settings*:
+** *Client type* (or *Client Protocol*, depending on your keycloak version): *OpenID Connect*
+** *Client ID*: *myclient*
+* *Capability config*:
+** *Client authentication* : *On*
+** *Authentication flow*: *Standard flow, Direct access grants*
+* *Login Settings*: For the *Valid Redirect URIs*, enter a * for now. We will come back to edit it later.
+
+. Once you hit *Save*, you should see a new tab called *Keys* appear.
+
+. Navigate to that tab and click on *Import*. 
+* Under *Archive format*, choose *PKCS12* from the dropdown. 
+* Under *Key alias*,
+enter the alias of the RSA key you just created. 
+* Under *Store password* enter the password of the keystore you created
+and click on the *Browse* button to import the keystore file. Note that when we created the keystore, we specified a key
+password and a keystore password. Here, we are only using the *keystore* password. While we set them both to be the same, they do not have to be.
+
+. Finally, create a user called *alice* whose *First name* is *Alice* and *Last name* is *Smith*. 
+
+. Next, navigate to the *Credentials* tab and create a password for Alice. Toggle the `Temporary` switch off, so you are
```

Comment:
I am stuck at this point and am unable to proceed.  I can not find the "Credentials" tab

Reply:
I have added more clear instructions for this.